src/instructlab/sdg/utils/parse_and_convert.py

```diff
@@ -0,0 +1,313 @@
+        )
+
+    parts = synth_example["output"].split("?", 1)
+    if len(parts) != 2:
+        logger.warning(f"Failed to split generated q&a: {synth_example['output']}")
+    return parts[0].strip() + "?" if len(parts) == 2 else ""
+
+
+# This is also a hack. See the comment above _get_question.
+def _get_response(synth_example: dict):
+    if "response" in synth_example:
+        return synth_example["response"]
+
+    if "output" not in synth_example:
+        raise utils.GenerateException(
+            f"Error: output not found in synth_example: {synth_example}"
+        )
+
+    parts = synth_example["output"].split("?", 1)
+    if len(parts) != 2:
+        logger.warning(f"Failed to split generated q&a: {synth_example['output']}")
+    return parts[1].strip() if len(parts) == 2 else parts[0].strip()
+
+
+def _convert_messages_to_legacy(sample: dict):
+    """
+    Convert new format messages to the legacy format 'system', 'user', and
+    'assistant' columns.
+
+    Note: We should remove this function in the future when standardize the
+    format to messages.
+    """
+    if len(sample["messages"]) < 3:
+        raise utils.GenerateException(
+            "Generated message cannot be converted to legacy format"
+        )
+    sample["system"] = _unescape(sample["messages"][0]["content"])
+    sample["user"] = _unescape(sample["messages"][1]["content"])
+    sample["assistant"] = _unescape(sample["messages"][2]["content"])
+
+    return sample
+
+
+def _convert_to_hack_fmt(sample: dict, sys_prompt: str):
+    """
+    Convert a sample dictionary to contain 'system', 'user', and 'assistant' columns.
+
+    Note: We should remove this function in the future when we resolve this issue and
+    standardize the format to messages.
+    """
+    # Create user query message
+    user_query = _unescape(_get_question(sample))
+    response = _unescape(_get_response(sample))
+    if "context" in sample:
+        user_query = f"{sample['context']}\n\n{user_query}"
+
+    sample["id"] = str(uuid.uuid4())
+    sample["system"] = sys_prompt
+    sample["user"] = user_query
+    sample["assistant"] = response
+
+    return sample
+
+
+def _convert_to_messages(sample: dict, sys_prompt: str):
+    """
+    Convert a sample dictionary to contain 'messages'
+    and 'metadata' columns required for training.
+    """
+    # Create user query message
+    user_query = _unescape(_get_question(sample))
+    response = _unescape(_get_response(sample))
+
+    sample["id"] = str(uuid.uuid4())
+    sample["messages"] = [
+        {"content": sys_prompt, "role": "system"},
+        {"content": user_query, "role": "user"},
+        {"content": response, "role": "assistant"},
+    ]
+
+    return sample
+
+
+def create_auxiliary_dataset(generated_dataset: Dataset):
+    if "dataset_type" not in generated_dataset.column_names:
+        return None
+    auxilary_path = "src/instructlab/sdg/config/kowledge/auxilary_instructions.yaml"
```

Comment:
"kowlege" typo ... and the file doesn't exist

Reply:
This feels like dead/outdated code to me, and I'm tempted to rip out all the auxiliary dataset bits in this PR to let someone else add them later with a follow-up and explanation of what it is and how it's used. I can't see anything that would cause this to actually run today, and it's code not exercised by any of the manual or automated testing I've performed. Looping in @aakankshaduggal and @shivchander in case they feel otherwise and can explain what this does.